module/VuFind/src/VuFind/Http/CachingDownloader.php

```diff
@@ -113,13 +135,11 @@ public function download(
         callable $validateCallback=null,
         callable $decodeCallback=null
     ) {
-        $cacheItemKey = md5($url);
-        foreach ($params as $paramKey => $paramValue) {
-            $cacheItemKey .= '#' . $paramKey . '§' . $paramValue;
-        }
+        $cache = $this->getCache();
+        $cacheItemKey = md5(http_build_query(array_merge([$url], $params)));
```

Comment:
I'm not sure if including the $url in the query makes sense; what about replacing `http_build_query(array_merge([$url], $params))` with `$url . http_build_query($params)`? I think that's a little simpler but should maintain the same degree of uniqueness.